lib/z80rules.1

```diff
@@ -6889,11 +6889,212 @@ $notcpu 8085
 	ld	de,%1	;const
 	ld	hl,%2	;const
 
+%title Reverse primary/secondary load
 %check 4 <= %2 <= 8
-	ld	hl,%1X
+	ld	hl,%1	;const
 	push	hl
 	call	l_gint%2sp	;
 	pop	de
 =
 	call	l_gint%eval(%2 2-)sp	;
 	ld	de,%1
+
+%title Optimise secondary load (perf)
+	ld	hl,(_%1)	X
+	ld	h,0
+	ex	de,hl
+	ld	hl,%2	;const
+=
+	ld	a,(_%1)
+	ld	e,a
+	ld	d,0
+	ld	hl,%2	;const
+
+%title Use 8 bit addition
+%check 0 <= %2 <= 256
+	ld	hl,(_%1)
+	ld	h,0
+	ld	bc,%2
+	add	hl,bc
+	ld	h,0
+=
+	ld	a,(_%1)
+	add	%eval(%2 256 + 256 %%)
+	ld	l,a
+	ld	h,0
+
+%title Load directly into a rather than via hl
+	ld	hl,_%1
+	ld	a,(hl)
+	pop	de
+	ld	(de),a
+	ld	hl,%2
+=
+	ld	a,(_%1)
+	pop	de
+	ld	(de),a
+	ld	hl,%2
+
+	ld	hl,_%1
+	ld	a,(hl)
+	pop	de
+	ld	(de),a
+	ld	l,a
+=
+	ld	a,(_%1)
+	pop	de
+	ld	(de),a
+	ld	l,a
+
+%notcpu gbz80
+	push	hl
+	ld	a,%1
+	pop	de
+=
+	ex	de,hl
+	ld	a,%1
+
+%check 0 <= %1 <= 255
+	ld	e,(hl)
+	ld	d,0
+	ld	hl,%1	;const
+	add	hl,sp
+	ld	l,(hl)
+	ld	h,0
+	add	hl,de
+	pop	de
+=
+	ld	a,(hl)
+	ld	hl,%1	;const
+	add	hl,sp
+	add	(hl)
+	ld	l,a
+	ld	a,h
+	adc	h
+	ld	h,a
```

Comment:
this is different? the H is not zero anymore (after `add hl,sp`).
imho this should be rather something like
```
 ld l,a
 ld h,0
 rl h
```
which is not clear win... also original code could do `ld h,d` instead of `ld h,0` to save 3T and one byte.
And finally not sure what the initial `%check` is for, as value in H is not preserved after add hl,sp, so it could be anything IMHO?

Reply:
It is wrong, thanks for the spot - all of my test cases obviously just cared about the lower 8 which is why they worked!